pubtools/_quay/signature_remover.py

```diff
@@ -249,3 +253,82 @@ def remove_repository_signatures(
             )
         else:
             LOG.info("No signatures need to be removed")
+
+    def remove_tag_signatures(
+        self,
+        reference,
+        pyxis_server,
+        pyxis_krb_principal,
+        pyxis_krb_ktfile=None,
+        exclude_by_claims=None,
+        remove_archs=None,
+    ):
+        """
+        Remove signatures of an image specified by a tag.
+
+        Source and multiarch images are supported. Signatures may be excluded from removal by
+        specifying a list of claim messages. If existing signature and a claim messages matches,
+        it is not removed, as the same exact signature would be recreated afterwards. Additionally,
+        it's also possible to only specify certain architectures whose signatures will be removed.
+        This option only has an effect when the image is a manifest list.
+
+        Args:
+            reference (str):
+                Image reference whose signatures are to be removed.
+            pyxis_server (str):
+                URL of the Pyxis service.
+            pyxis_krb_principal (str):
+                Kerberos principal to use for Pyxis authentication.
+            pyxis_krb_ktfile (str|None):
+                Path to Kerberos keytab file. Optional
+            exclude_by_claims ([dict]|None):
+                List of claim messages whose existing signature matches will not be removed.
+            remove_archs ([str]|None):
+                If specified and the reference is a multiarch image, only signatures of given
+                architectures will be eligible for removal.
+        """
+        if "@" in reference:
+            raise ValueError("Image, whose signatures are being removed must be specified by tag.")
+
+        full_repo, tag = reference.split(":", 1)
+        external_repo = get_external_container_repo_name(full_repo.split("/")[-1])
+        image_digests = []
+        repo_data = self.quay_api_client.get_repository_data(full_repo.split("/", 1)[-1])
+        # if image_id of the tag is specified, the image is V2S2 AKA source image
+        if repo_data["tags"][tag]["image_id"]:
```

Comment:
Getting image_digests should be skipped when reference is a not existing image.

Reply:
I skipped the rest of the method if the image doesn't exist